src/jit/lowerxarch.cpp

```diff
@@ -408,86 +408,28 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
 //
 void Lowering::LowerPutArgStk(GenTreePutArgStk* putArgStk)
 {
-#ifdef _TARGET_X86_
-    if (putArgStk->gtOp1->gtOper == GT_FIELD_LIST)
+    GenTree* src = putArgStk->gtGetOp1();
+
+    if (src->OperIs(GT_FIELD_LIST))
     {
+#ifdef _TARGET_X86_
         putArgStk->gtPutArgStkKind = GenTreePutArgStk::Kind::Invalid;
 
-        GenTreeFieldList* fieldList = putArgStk->gtOp1->AsFieldList();
+        GenTreeFieldList* fieldList = src->AsFieldList();
 
         // The code generator will push these fields in reverse order by offset. Reorder the list here s.t. the order
         // of uses is visible to LSRA.
-        unsigned          fieldCount = 0;
-        GenTreeFieldList* head       = nullptr;
-        for (GenTreeFieldList *current = fieldList, *next; current != nullptr; current = next)
-        {
-            next = current->Rest();
-
-            // First, insert the field node into the sorted list.
-            GenTreeFieldList* prev = nullptr;
-            for (GenTreeFieldList* cursor = head;; cursor = cursor->Rest())
```

Comment:
Sorting code was broken, it never updated `prev`. The reason why it worked is that the field list was already sorted by increasing offset so the bug here effectively turned this code into list reversal and we ended up getting the expected order. I don't see any reason to create field list with a different initial order so it seems preferable to just reverse the list and avoid the quadratic complexity concerns.

Reply:
Wow - glad you got this "sorted" ;-)